concrete/src/Package/Package.php

```diff
@@ -4,127 +4,296 @@
+
+    /**
+     * Path to the /concrete/packages directory relative to the web root.
+     *
+     * @var string
+     */
     protected $REL_DIR_PACKAGES = REL_DIR_PACKAGES;
 
     /**
-     * @var \Concrete\Core\Entity\Package
+     * Associated package entity.
+     *
+     * @var PackageEntity|null
      */
     protected $entity;
 
     /**
+     * The Application instance.
+     *
      * @var Application
      */
     protected $app;
 
     /**
-     * @var \Concrete\Core\Config\Repository\Liaison
+     * The database configuration liaison.
+     *
+     * @var Liaison|null
      */
     protected $config;
 
     /**
-     * @var \Concrete\Core\Config\Repository\Liaison
+     * The file configuration liaison.
+     *
+     * @var Liaison|null
      */
     protected $fileConfig;
 
     /**
      * @deprecated
-     * This will be set to FALSE in 8.1
-     * Additionally, if your package requires 8.0 or greater, it will be set to false automatically
-     * Whether to automatically map core extensions into the packages src/Concrete directory (and map them to Concrete\Package\MyPackage), or map the entire src/
-     * directory to Concrete\Package\MyPackage\Src (and automatically map core extensions
-     * to Concrete\Package\MyPackage\Src)
+     * Whether to automatically map core extensions into the packages src/Concrete directory (and map them to Concrete\Package\MyPackage),
+     * or map the entire src/ directory to Concrete\Package\MyPackage\Src
+     * (and automatically map core extensions to Concrete\Package\MyPackage\Src).
+     * This will be ALWAYS considered as FALSE if your package requires 8.0 or greater or if your package defines the pkgAutoloaderMapCoreExtensions property.
      *
      * @var bool
      */
     protected $pkgEnableLegacyNamespace = true;
 
     /**
-     * Array of location -> namespace autoloader entries for the package. Will automatically
-     * be added to the class loader. (e.g. array('src/PortlandLabs' => \PortlandLabs')).
+     * The custom autoloader prefixes to be automatically added to the class loader.
+     * Array keys are the locations (relative to the package directory).
+     * Array values are the paths (not relative to the package namespace).
      *
      * @var array
+     *
+     * @example ['src/PortlandLabs' => \PortlandLabs']
      */
     protected $pkgAutoloaderRegistries = [];
 
+    /**
+     * The minimum concrete5 version compatible with the package.
+     * Override this value according to the minimum required version for your package.
+     *
+     * @var string
+     */
     protected $appVersionRequired = '5.7.0';
 
+    /**
+     * Override this value setting it to true if your package clear all existing website content when it's being installed.
+     *
+     * @var bool
+     */
     protected $pkgAllowsFullContentSwap = false;
 
+    /**
+     * Override this value setting it to true if your package provide the file thumbnails.
```

Comment:
Override this value **and set** it to true if your package provide**s** the file thumbnails.